tests/spec.js

```diff
@@ -0,0 +1,152 @@
+            expect(resp.status).to.equal(200),
+            expect(resp.request.header).to.have.property('Authorization'),
+            expect(resp.request.header['Authorization']).to.have.match(/AWS4-HMAC-SHA256 Credential=a.*\/execute-api\/aws4_request, SignedHeaders=.*Signature=.*/i)
+          ]);
+        });
+    });
+
+    it('with provided values and adds sessionToken header', () => {
+      return request
+        .get('/')
+        .use(signRequest(service, { key: 'a', secret: 'b', sessionToken: 'c'}))
+        .then(resp => {
+          return Promise.all([
+            expect(resp.status).to.equal(200),
+            expect(resp.request.header).to.have.property('Authorization'),
+            expect(resp.request.header['Authorization']).to.have.match(/AWS4-HMAC-SHA256 Credential=a.*\/execute-api\/aws4_request, SignedHeaders=.*x-amz-security-token, Signature=.*/i),
+            expect(resp.request.header).to.have.property('X-Amz-Security-Token'),
+            expect(resp.request.header['X-Amz-Security-Token']).to.equal('c')
+          ]);
+        });
+    });
+  });
+
+  describe('signs the POST request', () => {
+    let req;
+    const testPostData = {json: 'data'};
+    beforeEach(done => {
+      req = nock('http://localhost', {
+        reqheaders: {
+          Authorization: /AWS4-HMAC-SHA256 Credential=.*\/execute-api\/aws4_request, SignedHeaders=.*Signature=.*/i,
+          'X-Amz-Date': /.*/i
+        }
+      })
+      .post('/', testPostData)
+      .reply(200);
+
+      done();
+    });
+
+    afterEach(done => {
+      req.done();
+      done();
+    });
+
+    it('with default values', () => {
+      return request
+        .post('/')
+        .use(signRequest(service))
+        .send(testPostData)
+        .then(resp => {
+          return Promise.all([
+            expect(resp.status).to.equal(200),
+            expect(resp.request.header).to.have.property('Authorization'),
+            expect(resp.request.header['Authorization']).to.have.match(/AWS4-HMAC-SHA256 Credential=.*\/execute-api\/aws4_request, SignedHeaders=.*Signature=.*/i)
+          ]);
+        });
+    });
+
+    it('with provided values', () => {
+      return request
+        .post('/')
+        .use(signRequest(service, { key: 'a', secret: 'b'}))
+        .send(testPostData)
+        .then(resp => {
+          return Promise.all([
+            expect(resp.status).to.equal(200),
+            expect(resp.request.header).to.have.property('Authorization'),
+            expect(resp.request.header['Authorization']).to.have.match(/AWS4-HMAC-SHA256 Credential=a.*\/execute-api\/aws4_request, SignedHeaders=.*Signature=.*/i)
+          ]);
+        });
+    });
+
+    it('with provided values and adds sessionToken header', () => {
+      return request
+        .post('/')
+        .use(signRequest(service, { key: 'a', secret: 'b', sessionToken: 'c'}))
+        .send(testPostData)
+        .then(resp => {
+          return Promise.all([
```

Comment:
Yah, not sure if I even need to use for any of these, think I got used to writing the tests this way from attributes :P. I'll clean up